site/docs/patterns/forms.mdx

```diff
@@ -0,0 +1,839 @@
+    </ul>
+</div>
+
+<div class="guideline-dont" style="background: var(--color-error-light); padding: var(--spacing-s); margin-bottom: var(--spacing-layout-2-xs);">
+    <div class="guideline-dont-label" style="color:var(--color-error); margin-bottom:var(--spacing-s);"><IconError style={{ marginRight: 'var(--spacing-3-xs)', verticalAlign: 'middle'}} size="s"/><strong>Don't</strong></div>
+    <p><strong>An icon <em>must</em> not be used to perform an action.</strong> Use an action button instead.</p>
+</div>
+
+---
+
+### 5. Action button
+Some HDS form controls have actionable elements for **executing commands or other actions** within the form element. They provide an additional functionality to the form control for example to:
+- Execute a search with inputted value.
+- Clearing all selections in multi-select dropdowns.
+
+Although the action button is visually similar to an icon, the implementation is different. 
+
+<Image src="../../static/patterns/form/form-elements-action@2x.png" alt="Form control icon" style="max-width:376px;" viewable />
+
+<div class="guideline-dont" style="background: var(--color-error-light); padding: var(--spacing-s); margin-bottom: var(--spacing-layout-2-xs);">
+    <div class="guideline-dont-label" style="color:var(--color-error); margin-bottom:var(--spacing-s);"><IconError style={{ marginRight: 'var(--spacing-3-xs)', verticalAlign: 'middle'}} size="s"/><strong>Don't</strong></div>
+    <p>In HDS form components, action buttons are used for predefined purposes and <em>should</em> not be customised.</p>
+</div>
+
+---
+
+
+## Interacting with controls
+
+### Active and disabled
+Form controls are either *active* or *disabled* depending on for example selections made in previous controls in the form, or user permissions. 
+
+- **Active form controls can be focused and given a value.** Form components are active by default
+- **Disabled form controls cannot be clicked or focused with keyboard.**
+    - The disabled attribute is be set for a form control to keep a user from using it until some other condition has been met (for example a text input can be disabled before a checkbox is selected).
+
+<Image src="../../static/patterns/form/form-states-active-disabled@2x.png" alt="Active and disabled form controls" style="max-width:376px;" viewable /> 
+
+<div class="guideline-do" style="background: var(--color-success-light); padding: var(--spacing-s); margin-bottom: var(--spacing-layout-2-xs);">
+    <div class="guideline-do-label" style="color:var(--color-success); margin-bottom:var(--spacing-s);"><IconCheck style={{ marginRight: 'var(--spacing-3-xs)', verticalAlign: 'middle'}} size="s"/><strong>Do</strong></div>
+    <ul style="margin-bottom:0;">
+        <li>Disabled form controls can also be hidden from view, but in most situations all fields, even the disabled ones should be shown to the user.
```

Comment:
> Disabled form controls can also be hidden from view

Yleensä sanotaan myös niin, että jos käyttäjällä ei ole mitään keinoa saada kenttää aktivoiduksi, niin sitten sitä ei välttämättä kannata näyttää ollenkaan (jollei sille ole jotain muuta hyvää syytä). En tiedä kannattaisiko tämä ottaa meidän ohjeistuksiin mukaan.

Reply:
Tein lisäyksen:
As a rule of thumb, a disabled field should only be hidden, if the user
- has no way of activating the field,
- does not have permission to change it’s value.